stellarphot/settings/tests/test_custom_widgets.py

```diff
@@ -187,6 +188,36 @@ def test_choosing_different_item_updates_display(self, tmp_path):
         # The item widget should now have the values of the second observatory
         assert Observatory(**choose_or_make_new._item_widget.value) == observatory2
 
+    def test_passband_map_buttons_are_disabled_or_enabled(self, tmp_path):
+        # When an existing PassbandMap is selected the add/remove buttons
+        # for individual rows should not be displayed.
+        saved = SavedSettings(_testing_path=tmp_path)
+        passband_map = PassbandMap(**DEFAULT_PASSBAND_MAP)
+        saved.add_item(passband_map)
+
+        choose_or_make_new = ChooseOrMakeNew("passband_map", _testing_path=tmp_path)
+
+        # There is no great way to get to the ItemBox widget that contains and controls
+        # the add/remove buttons, so we keep going down through widget children until we
+        # get to an ItemBox and then check that the buttons are disabled.
+        # Recursion is the easiest way to do that, so recurse we will..
```

Comment:
Wow, an actual use for recursion that isn't factorials or the Fibonacci sequence or binary search...